test/server/cards/02_SHD/units/EnterprisingLackeys.spec.ts

```diff
@@ -0,0 +1,56 @@
+describe('Enterprising Lackeys', function() {
+    integration(function(contextRef) {
+        describe('Enterprising Lackeys\'s when defeated ability', function() {
+            beforeEach(function () {
+                contextRef.setupTest({
+                    phase: 'action',
+                    player1: {
+                        groundArena: ['enterprising-lackeys'],
+                        resources: ['wampa', 'battlefield-marine', 'wild-rancor', 'protector', 'devotion', 'restored-arc170']
+                    },
+                    player2: {
+                        hand: ['vanquish'],
+                        hasInitiative: true,
+                    }
+                });
+            });
+
+            it('should defeat a resource and put this card as resource', function () {
+                const { context } = contextRef;
+
+                context.player2.clickCard(context.vanquish);
+
+                // select a resource to defeat
+                expect(context.player1).toBeAbleToSelectExactly([context.wampa, context.battlefieldMarine, context.wildRancor, context.protector, context.devotion, context.restoredArc170]);
+
+                expect(context.player1).toHavePassAbilityButton();
+                expect(context.player1).toHaveChooseNoTargetButton();
+
+                context.player1.clickCard(context.wampa);
+
+                // wampa should be defeated and lackeys should be in resource
+                expect(context.wampa).toBeInZone('discard');
+                expect(context.enterprisingLackeys).toBeInZone('resource');
+                expect(context.player1).toBeActivePlayer();
```

Comment:
Frugghi just added logic so that if a resource is moved out of the resource zone when it is not exhausted, the game will attempt to exhaust it and ready another resource if possible. That way the player isn't losing a ready resource for stuff like this or Smuggle.

Can we test this here by exhausting one resource that isn't being defeated and confirm that the available resource count stays the same?